eng/native/build-commons.sh

```diff
@@ -117,7 +117,7 @@ build_native()
             scan_build=scan-build
         fi
 
-        nextCommand="\"$__RepoRootDir/eng/native/gen-buildsys.sh\" \"$cmakeDir\" \"$intermediatesDir\" $platformArch $__Compiler \"$__CompilerMajorVersion\" \"$__CompilerMinorVersion\" $__BuildType \"$generator\" $scan_build $cmakeArgs"
+        nextCommand="\"$__RepoRootDir/eng/native/gen-buildsys.sh\" \"$cmakeDir\" \"$intermediatesDir\" $platformArch $__Compiler $__BuildType \"$generator\" $scan_build $cmakeArgs"
```

Comment:
This (and the changes in `gen-buildsys.sh`) was the main breaking change, because we are using positional arguments in internal scripts as opposed to named ones. Before, we were splitting up `-clang12.2` into `"clang" "12" "2"` in this script, and passing it to init-compiler.sh. Now we pass the raw value `clang12.2` down to init-compiler.sh, which splits the compiler name and version parts autonomously.

Related to these user-facing scenarios:
```sh
# LLVM toolchain is default
# uses latest available clang by default. if someone has `clang-8`, `clang-9` and `clang-10`,
# init-compiler picks up clang 10
./build.sh

# same here, find latest version of clang on the system
./build.sh -clang

# find clang 12 explicitly on the machine, and error out if it was not found; no fallback.
# this was an explicit decision made with JanV in 2020. so we don't accidentally build
# with wrong toolchain in official builds.
./build.sh -clang12 / -clang-12

# similarly, for GNU toolchain
# find latest of gcc and use that to compile
./build.sh -gcc

# find gcc 11 on machine, if not found, error out
./build.sh -gcc11 / -gcc-11
```

cc @mikem8361, @hoyosjs